deltacat/tests/compactor/test_compaction_session_incremental.py

```diff
@@ -0,0 +1,341 @@
+    return source_table_stream_after_committed, destination_table_stream
+
+
+@pytest.mark.parametrize(
+    [
+        "test_name",
+        "primary_keys_param",
+        "sort_keys_param",
+        "partition_keys_param",
+        "column_names_param",
+        "arrow_arrays_param",
+        "rebase_source_partition_locator_param",
+        "partition_values_param",
+        "expected_result",
+        "validation_callback_func",
+        "validation_callback_func_kwargs",
+        "cleanup_prev_table",
+        "use_prev_compacted",
+        "create_placement_group_param",
+        "records_per_compacted_file_param",
+        "hash_bucket_count_param",
+    ],
+    [
+        (
+            test_name,
+            primary_keys_param,
+            sort_keys_param,
+            partition_keys_param,
+            column_names_param,
+            arrow_arrays_param,
+            rebase_source_partition_locator_param,
+            partition_values_param,
+            expected_result,
+            validation_callback_func,
+            validation_callback_func_kwargs,
+            cleanup_prev_table,
+            use_prev_compacted,
+            create_placement_group_param,
+            records_per_compacted_file_param,
+            hash_bucket_count_param,
+        )
+        for test_name, (
+            primary_keys_param,
+            sort_keys_param,
+            partition_keys_param,
+            column_names_param,
+            arrow_arrays_param,
+            rebase_source_partition_locator_param,
+            partition_values_param,
+            expected_result,
+            validation_callback_func,
+            validation_callback_func_kwargs,
+            cleanup_prev_table,
+            use_prev_compacted,
+            create_placement_group_param,
+            records_per_compacted_file_param,
+            hash_bucket_count_param,
+        ) in INCREMENTAL_TEST_CASES.items()
+    ],
+    ids=[test_name for test_name in INCREMENTAL_TEST_CASES],
+)
+def test_compact_partition_incremental(
+    request,
+    s3_resource: ServiceResource,
+    ds_mock_kwargs: Dict[str, Any],
+    compaction_artifacts_s3_bucket: None,
+    test_name: str,
+    primary_keys_param: Set[str],
+    sort_keys_param,
+    partition_keys_param,
+    column_names_param: List[str],
+    arrow_arrays_param: List[pa.Array],
+    rebase_source_partition_locator_param,
+    partition_values_param,
+    expected_result,
+    validation_callback_func,  # use and implement if you want to run additional validations apart from the ones in the test
+    validation_callback_func_kwargs,
+    cleanup_prev_table,
+    use_prev_compacted,
+    create_placement_group_param,
+    records_per_compacted_file_param,
+    hash_bucket_count_param,
+):
+
+    """
+    TODO Test Cases:
+    1. incremental w/wout round completion file
+    2. Backfill w/wout round completion
+    3. Rebase w/wout round completion file
+    4. Rebase then incremental (use same round completion file)
```

Comment:
As far as this file is concerned, it is complete. We can move this TODO to testcases.py.